services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_service_specs/sidecar.py

```diff
@@ -78,6 +78,10 @@ def _get_environment_variables(
         "S3_SECRET_KEY": r_clone_settings.R_CLONE_S3.S3_SECRET_KEY,
         "S3_SECURE": f"{r_clone_settings.R_CLONE_S3.S3_SECURE}",
         "SC_BOOT_MODE": f"{app_settings.DYNAMIC_SERVICES.DYNAMIC_SIDECAR.DYNAMIC_SIDECAR_SC_BOOT_MODE}",
+        "SSL_CERT_FILE": app_settings.DIRECTOR_V2_SELF_SIGNED_SSL_FILENAME,
```

Comment:
OK I see where the issue is. You are passing an env var to the dynamic sidecar via the docker service spec. This needs to be of type `dict[str, str]`.

You now have the issue of: `how do I deal with None or missing values?`
The dy-sidecar is using pedantic. For it, it is OK to just serialise `None` object to `"None"` string, and this will be interpreted correctly.

In your case, you are passing this value to some other program that is ran before the dy-sidecar is started. I would suggest not adding this env var if the value of `app_settings.DIRECTOR_V2_SELF_SIGNED_SSL_FILENAME` is `None`.

Reply:
Also no further changes are required here since `SSL_CERT_FILE` is always going to be a valid string